applications/fishing-map/src/features/workspace/heatmaps/HeatmapFilters.tsx

```diff
@@ -18,15 +26,28 @@ function Filters({ dataview }: FiltersProps): React.ReactElement {
 
   const sourceOptions = getSourcesOptionsInDataview(dataview)
   const sourcesSelected = getSourcesSelectedInDataview(dataview)
-
-  const fishingFiltersOptions = getFlagsByIds(dataview.config?.filters || [])
+  const gearTypeOptions = getCommonGearTypesInDataview(dataview)
+  const gearTypeSelected = gearTypeOptions?.filter((geartype) =>
+    dataview.config?.filters?.geartype?.includes(geartype.id)
+  )
+  const fishingFiltersOptions = getFlagsByIds(dataview.config?.filters?.flag || [])
   const flags = useMemo(getFlags, [])
 
+  const datasetsWithoutgGeartype = getNotSupportedGearTypesDatasets(dataview)
   const onSelectSourceClick: MultiSelectOnChange = (source) => {
+    const datasets = [...(dataview.config?.datasets || []), source.id]
+    // We have to remove the geartype if it is not supported by the datasets selecion
+    const gearTypeOptions = getCommonGearTypesInDataview({ ...dataview, datasets })
+    const supportGearTypeFilter = gearTypeOptions.length > 0
+    const filters = dataview.config?.filters ? { ...dataview.config.filters } : {}
+    if (!supportGearTypeFilter && filters['geartype']) {
```

Comment:
Thinking about a scenario where the user selects one source and for the gear types filters selects two: Drifting longlines and Set longlines. And then selects another source in addition than only supports Set longlines. Is the gear type filters selection updated?

Reply:
Good catch! I'll fix it, thanks for spotting it out

Reply:
Fixed in https://github.com/GlobalFishingWatch/frontend/commit/24b604f3acb9e877c986c2f7f680d737ee5b016f
I had to mock VMS Indonesia dataset to include `drifting_longlines` and `tuna_purse_seines` but now the selection is updated with the options both datasets shares, an example as it is nos possible to test without the mock:

https://user-images.githubusercontent.com/10500650/105324930-52783e00-5bcc-11eb-83fd-3fb087281138.mp4